src/borg/archive.py

```diff
@@ -1877,8 +1877,11 @@ def valid_item(obj):
                 return
         num_archives = len(archive_infos)
 
+        pi = ProgressIndicatorPercent(total=num_archives, msg='Checking archives %3.1f%%', step=0.1,
+                                      msgid='archive.robust_iterator')
         with cache_if_remote(self.repository) as repository:
             for i, info in enumerate(archive_infos):
+                pi.show(i)
```

Comment:
usually showing the progress indicator depends on `--progress` being given, see other places where it is used.

Reply:
While I indeed have not fully analyzed the workings of this, I did observe that the equivalent code for the "Checknig segments" progress indicator does not have any such conditional either, the code does work as desired (showing progress output when `--progress` is given and not showing any output if not) and I do see that the `ProgressIndicatorBase` class uses the `borg.output.progress` logger, so I would assume that logger is only configured to output things if `--progress` was given. :slightly_smiling_face:

Reply:
Right. Must have had some code from the past in my head.

Strange though, that some places in the code still check `if ... progress:` before calling `.show()`.